crates/wast/src/component/import.rs

```diff
@@ -30,17 +67,175 @@ pub enum ComponentExternName<'a> {
     Kebab(&'a str),
     /// This is an interface import where the string is an ID.
     Interface(&'a str),
+    /// This is an implementation import where the string is an ID.
+    Implementation(ImplementationImport<'a>),
+}
+
+/// The different names that can be assigned to component import
+#[derive(Debug, Copy, Clone)]
+pub enum ImplementationImport<'a> {
+  /// External Url
+  Url(ImportMetadata<'a>),
+  /// Relative path
+  Relative(ImportMetadata<'a>),
+  /// Just Integrity
+  Naked(ImportMetadata<'a>),
+  /// Locked Registry Import
+  Locked(ImportMetadata<'a>),
+  /// Unlocked Registry Import
+  Unlocked(ImportMetadata<'a>),
```

Comment:
Additionally, like wasmparser/wasm-encoder, I think each of these variants should have a specific payload just to that one variant.